[GTK][WPE] Remove common code to get the a11y bus address from PlatformDisplay

carlosgcampos opened the pull request:
#### e21334567540b56505d57741166cc20796dfdb08
<pre>
[GTK][WPE] Remove common code to get the a11y bus address from PlatformDisplay
<a href="https://bugs.webkit.org/show_bug.cgi?id=278164">https://bugs.webkit.org/show_bug.cgi?id=278164</a>

Reviewed by Adrian Perez de Castro.

Keeping only the GTK platform specific code that will be moved in a
follow up. The a11y bus address is now handled by WebProcessPool, that
passes both the host and the sandboxed dbus addresses to the launcher
as extra initialization data of the launch options.

* Source/WTF/wtf/glib/Sandbox.cpp:
(WTF::sandboxedUserRuntimeDirectory):
(WTF::sandboxedAccessibilityBusAddress): Deleted.
(WTF::setSandboxedAccessibilityBusAddress): Deleted.
* Source/WTF/wtf/glib/Sandbox.h:
* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::PlatformDisplay):
(WebCore::PlatformDisplay::accessibilityBusAddress const):
* Source/WebCore/platform/graphics/PlatformDisplay.h:
(WebCore::PlatformDisplay::platformAccessibilityBusAddress const): Deleted.
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.cpp:
(WebCore::PlatformDisplayX11::accessibilityBusAddress const):
(WebCore::PlatformDisplayX11::platformAccessibilityBusAddress const): Deleted.
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.h:
* Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp:
(WebKit::bindDBusSession):
(WebKit::bindA11y):
(WebKit::bubblewrapSpawn):
(WebKit::dbusProxyDirectory): Deleted.
* Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.cpp:
(WebKit::XDGDBusProxy::accessibilityProxy):
(WebKit::XDGDBusProxy::launch):
* Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.h:
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::platformInitializeWebProcess):
(WebKit::WebProcessPool::setSandboxEnabled):
(WebKit::queryAccessibilityBusAddress):
(WebKit::WebProcessPool::accessibilityBusAddress const):
(WebKit::WebProcessPool::sandboxedAccessibilityBusAddress const):
* Source/WebKit/UIProcess/glib/WebProcessProxyGLib.cpp:
(WebKit::WebProcessProxy::platformGetLaunchOptions):

Canonical link: <a href="https://commits.webkit.org/282339@main">https://commits.webkit.org/282339@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/85467a1f583f9d1b691fc1b5ff34dbeebdaffc44

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/62879 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42235 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15475 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/66899 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/13483 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/64999 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/49922 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/13767 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/50701 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9309 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/65948 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39258 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54464 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31382 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/35950 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/11797 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12359 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/55989 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57498 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12127 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/68594 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/62122 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/6825 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/11762 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58016 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/6857 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54524 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58209 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/5697 "Passed tests") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/83885 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9471 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38055 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/14761 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/39135 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40246 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/38877 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebCore/platform/graphics/PlatformDisplay.cpp:
```diff
@@ -450,46 +445,16 @@ const Vector<PlatformDisplay::DMABufFormat>& PlatformDisplay::dmabufFormats()
-            }
-        }
+#if USE(GTK4)
+    if (m_sharedDisplay) {
+        if (const char* atspiBusAddress = static_cast<const char*>(g_object_get_data(G_OBJECT(m_sharedDisplay.get()), "-gtk-atspi-bus-address")))
+            return String::fromUTF8(atspiBusAddress);
     }
```
I don't like that the result will be different depending on whether or not `m_sharedDisplay` has been initialized. Would be better to create the shared display here if needed.

carlosgcampos replied:
m_sharedDisplay is initialized in the constructor, but it can become nullptr if the GdkDisplay is closed (unlikely). So, the result will always be the same.

mcatanzaro reviewed (commented):
Have you considered really completely removing `PlatformDisplay::accessibilityBusAddress`? I think it would be a lot simpler to move all the code to `WebProcessPool`, even the stuff that deals with "-gtk-atspi-bus-address" and the `XGetWindowProperty`. Having all the code in one place would make it easier to follow.

carlosgcampos committed:
[GTK][WPE] Remove common code to get the a11y bus address from PlatformDisplay
https://bugs.webkit.org/show_bug.cgi?id=278164

Reviewed by Adrian Perez de Castro.

Keeping only the GTK platform specific code that will be moved in a
follow up. The a11y bus address is now handled by WebProcessPool, that
passes both the host and the sandboxed dbus addresses to the launcher
as extra initialization data of the launch options.

* Source/WTF/wtf/glib/Sandbox.cpp:
(WTF::sandboxedUserRuntimeDirectory):
(WTF::sandboxedAccessibilityBusAddress): Deleted.
(WTF::setSandboxedAccessibilityBusAddress): Deleted.
* Source/WTF/wtf/glib/Sandbox.h:
* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::PlatformDisplay):
(WebCore::PlatformDisplay::accessibilityBusAddress const):
* Source/WebCore/platform/graphics/PlatformDisplay.h:
(WebCore::PlatformDisplay::platformAccessibilityBusAddress const): Deleted.
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.cpp:
(WebCore::PlatformDisplayX11::accessibilityBusAddress const):
(WebCore::PlatformDisplayX11::platformAccessibilityBusAddress const): Deleted.
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.h:
* Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp:
(WebKit::bindDBusSession):
(WebKit::bindA11y):
(WebKit::bubblewrapSpawn):
(WebKit::dbusProxyDirectory): Deleted.
* Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.cpp:
(WebKit::XDGDBusProxy::accessibilityProxy):
(WebKit::XDGDBusProxy::launch):
* Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.h:
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::platformInitializeWebProcess):
(WebKit::WebProcessPool::setSandboxEnabled):
(WebKit::queryAccessibilityBusAddress):
(WebKit::WebProcessPool::accessibilityBusAddress const):
(WebKit::WebProcessPool::sandboxedAccessibilityBusAddress const):
* Source/WebKit/UIProcess/glib/WebProcessProxyGLib.cpp:
(WebKit::WebProcessProxy::platformGetLaunchOptions):

Canonical link: https://commits.webkit.org/282339@main
